mod/node-core/pkg/components/service_registry.go

```diff
@@ -30,45 +30,35 @@ import (
 // ServiceRegistryInput is the input for the service registry provider.
 type ServiceRegistryInput struct {
 	depinject.In
-	ABCIService           *ABCIMiddleware
-	BlockBroker           *BlockBroker
-	BlockStoreService     *BlockStoreService
-	ChainService          *ChainService
-	DAService             *DAService
-	DBManager             *DBManager
-	DepositService        *DepositService
-	EngineClient          *EngineClient
-	GenesisBroker         *GenesisBroker
-	Logger                log.Logger
-	NodeAPIServer         *NodeAPIServer
-	ReportingService      *ReportingService
-	SidecarsBroker        *SidecarsBroker
-	SlotBroker            *SlotBroker
-	TelemetrySink         *metrics.TelemetrySink
-	ValidatorService      *ValidatorService
-	ValidatorUpdateBroker *ValidatorUpdateBroker
+	BlockStoreService *BlockStoreService
+	ChainService      *ChainService
+	DAService         *DAService
+	DBManager         *DBManager
+	DepositService    *DepositService
+	EngineClient      *EngineClient
+	Logger            log.Logger
+	NodeAPIServer     *NodeAPIServer
+	ReportingService  *ReportingService
+	TelemetrySink     *metrics.TelemetrySink
+	ValidatorService  *ValidatorService
+	Dispatcher        *Dispatcher
 }
 
 // ProvideServiceRegistry is the depinject provider for the service registry.
 func ProvideServiceRegistry(
 	in ServiceRegistryInput,
 ) *service.Registry {
 	return service.NewRegistry(
+		in.Dispatcher,
```

Comment:
should the service registry force the requirement of having a dispatcher? in theory (very loose in theory) someone using as an sdk doesn't need to adhere to any of our async service logic